connectors/byoei.py

```diff
@@ -127,12 +135,29 @@ async def _batch_bulk(self, operations):
                         if "error" in data:
                             logger.error(f"operation {op} failed, {data['error']}")
                             raise Exception(data["error"]["reason"])
+
+            for op in operations:
+                if OP_INDEX in op or OP_UPSERT in op:
+                    self.indexed_document_count += 1
+                elif OP_DELETE in op:
+                    self.deleted_document_count += 1
+                elif "doc" in op:  # update payload
+                    self.indexed_document_volume += get_byte_size(op["doc"])
+                else:  # index payload
+                    self.indexed_document_volume += get_byte_size(op)
+
         finally:
             self.bulk_time += time.time() - start
 
         return res
 
     async def run(self):
+        try:
+            await self._run()
+        except asyncio.CancelledError:
+            logger.info("Task is canceled, stop Bulker...")
```

Comment:
Do we need to have a `finally` block here that disposes of used resources/tasks?